infrastructure/code/code_html.go

```diff
@@ -30,172 +31,157 @@ import (
 	"github.com/snyk/snyk-ls/domain/snyk"
 )
 
+type IgnoreDetail struct {
+	Label string
+	Value string
+}
+
+type DataFlowItem struct {
+	Number         int
+	FilePath       string
+	StartLine      int
+	EndLine        int
+	StartCharacter int
+	EndCharacter   int
+	FileName       string
+	Content        string
+	StartLineValue int
+}
+
+type ExampleCommit struct {
+	CommitURL string
+	IconSVG   template.HTML
+	RepoName  string
+	DiffHTML  template.HTML
+}
+
```

Comment:
There is already structs for DataFlowElements and ExampleCommits. Should we just extend them? Or doesn't it make sense?

Reply:
The idea is that I'm preparing the data that will be injected in the template:

```html
<tbody>
  {{range .DataFlowTable}}
  <tr class="data-flow-row">
    <td class="data-flow-number">{{.Number}}</td>
    <td class="data-flow-clickable-row" file-path="{{.FilePath}}" start-line="{{.StartLine}}"
      end-line="{{.EndLine}}" start-character="{{.StartCharacter}}" end-character="{{.EndCharacter}}">
      {{.FileName}}:{{.StartLineValue}}
    </td>
    <td class="data-flow-delimiter">|</td>
    <td class="data-flow-text">{{.Content}}</td>
  </tr>
  {{end}}
</tbody>
```

and the local `struct` for `DataFlowItem` is specific for that template:
```go
DataFlowItem{
	Number:         i + 1,
	FilePath:       flow.FilePath,
	StartLine:      flow.FlowRange.Start.Line,
	EndLine:        flow.FlowRange.End.Line,
	StartCharacter: flow.FlowRange.Start.Character,
	EndCharacter:   flow.FlowRange.End.Character,
	FileName:       filepath.Base(flow.FilePath),
	Content:        flow.Content,
	StartLineValue: flow.FlowRange.Start.Line + 1,
}
```

Does this make sense? Was it what you meat?

Reply:
I was wondering if we need the additional type. On the other hand, it doesn't hurt to have it.